src/stores/spaces/SpaceStore.ts

```diff
@@ -719,31 +686,40 @@ export class SpaceStoreClass extends AsyncStoreWithClient<IState> {
                 const expandedRoomIds = new Set(Array.from(roomIds).flatMap(roomId => {
                     return this.matrixClient.getRoomUpgradeHistory(roomId, true).map(r => r.roomId);
                 }));
+
                 this.spaceFilteredRooms.set(spaceId, expandedRoomIds);
+
                 this.spaceFilteredUsers.set(spaceId, userIds);
                 return [expandedRoomIds, userIds];
             };
 
-            fn(s.roomId, new Set());
+            traverseSpace(s.roomId, new Set());
         });
 
-        const roomDiff = mapDiff(oldFilteredRooms, this.spaceFilteredRooms);
-        const userDiff = mapDiff(oldFilteredUsers, this.spaceFilteredUsers);
+        const roomDiff = mapDiff(prevRoomsBySpace, this.spaceFilteredRooms);
+        const userDiff = mapDiff(prevUsersBySpace, this.spaceFilteredUsers);
+        const spaceDiff = mapDiff(prevChildSpacesBySpace, this.childSpacesBySpace);
         // filter out keys which changed by reference only by checking whether the sets differ
         const roomsChanged = roomDiff.changed.filter(k => {
-            return setHasDiff(oldFilteredRooms.get(k), this.spaceFilteredRooms.get(k));
+            return setHasDiff(prevRoomsBySpace.get(k), this.spaceFilteredRooms.get(k));
         });
         const usersChanged = userDiff.changed.filter(k => {
-            return setHasDiff(oldFilteredUsers.get(k), this.spaceFilteredUsers.get(k));
+            return setHasDiff(prevUsersBySpace.get(k), this.spaceFilteredUsers.get(k));
+        });
+        const spacesChanged = spaceDiff.changed.filter(k => {
+            return setHasDiff(prevChildSpacesBySpace.get(k), this.childSpacesBySpace.get(k));
         });
 
         const changeSet = new Set([
             ...roomDiff.added,
             ...userDiff.added,
+            ...spaceDiff.added,
             ...roomDiff.removed,
             ...userDiff.removed,
+            ...spaceDiff.removed,
             ...roomsChanged,
             ...usersChanged,
+            ...spacesChanged,
         ]);
```

Comment:
If a subspace's direct descendent room list changes this seems like it'll no longer emit an update on its parent spaces (recursing) which is a behavioural change

Reply:
Fixed this using the `parentMap`, here and in `onMembersUpdate`.

Thanks for all your reviews here!